test/test_torch.py

```diff
@@ -6405,6 +6405,13 @@ def test_diagonal(self, device):
         expected = torch.diag(x, 17)
         self.assertEqual(result, expected)
 
+        # test bool tensor
+        a = torch.tensor([True, False, True], device=device)
+        expected = torch.tensor([[True, False, False],
+                                 [False, False, False],
+                                 [False, False, True]])
+        self.assertEqual(a.diag(), expected)
```

Comment:
Shouldn't this be in `test_diag` and not in `test_diagonal`?

Reply:
Also, it seems to me like this test should be dtype-generic.

Reply:
Added additional test as rand in not supported for bool tensor.